libs/remix-core-plugin/src/lib/openzeppelin-proxy.ts

```diff
@@ -118,51 +137,58 @@ export class OpenZeppelinProxy extends Plugin {
         dataHex: dataHex,
         funAbi: GETUUPSProxyVersionAbi,
         contractName: proxyName,
-        funArgs: []
+        funArgs: [],
       },
-      useCall: true
+      useCall: true,
     }
     // re-use implementation contract's ABI for UI display in udapp and change name to proxy name.
     newImplementationContractObject.contractName = newImplementationContractObject.name
     newImplementationContractObject.implementationAddress = newImplAddress
     newImplementationContractObject.name = proxyName
-    
-    await this.blockchain.runTx(args, () => { }, () => { }, () => { }, async (error, txResult, _address, returnValue) => {
-      if (error) {
-        throw new Error(`error: ${error.message ? error.message : error}`)
-      }
-      const response = txFormat.decodeResponse(returnValue, GETUUPSProxyVersionAbi)
-      const version = response[0].split('string: ')[1]
-      // check if version is >= 5.0.0
-      if (semver.gte(version, '5.0.0')) {
-        const fnData = await this.blockchain.getEncodedFunctionHex([newImplAddress, "0x"], UUPSupgradeToAndCallAbi)
-
-        const data = {
-          contractABI: UUPSABI,
-          contractName: proxyName,
-          funAbi: UUPSupgradeToAndCallAbi,
-          funArgs: [newImplAddress, "0x"],
-          linkReferences: {},
-          dataHex: fnData.replace('0x', '')
-        }
 
-        this.blockchain.upgradeProxy(proxyAddress, newImplAddress, data, newImplementationContractObject)
-      } else {
-        const fnData = await this.blockchain.getEncodedFunctionHex([newImplAddress], UUPSupgradeAbi)
-        const proxyName = 'ERC1967Proxy'
-        const data = {
-          contractABI: UUPSABI,
-          contractName: proxyName,
-          funAbi: UUPSupgradeAbi,
-          funArgs: [newImplAddress],
-          linkReferences: {},
-          dataHex: fnData.replace('0x', '')
+    await this.blockchain.runTx(
+      args,
+      () => {},
+      () => {},
+      () => {},
+      async (error, txResult, _address, returnValue) => {
```

Comment:
this part didn't change too right? except of the new logs in the terminal.

Reply:
no it's wrapped in a callback from the tx that gets the version of the erc

Reply:
so it makes two calls depending on the version